src/modules/battery_status/analog_battery.cpp

```diff
@@ -68,19 +69,51 @@ AnalogBattery::AnalogBattery(int index, ModuleParams *parent, const int sample_i
 
 	snprintf(param_name, sizeof(param_name), "BAT%d_I_CHANNEL", index);
 	_analog_param_handles.i_channel = param_find(param_name);
+
+	snprintf(param_name, sizeof(param_name), "BAT%d_T_CHANNEL", index);
+	_analog_param_handles.t_channel = param_find(param_name);
+
+	snprintf(param_name, sizeof(param_name), "BAT%d_T_R_PU", index);
+	_analog_param_handles.t_r_pu = param_find(param_name);
+
+	snprintf(param_name, sizeof(param_name), "BAT%d_T_R_25C", index);
+	_analog_param_handles.t_r_25c = param_find(param_name);
+
+	snprintf(param_name, sizeof(param_name), "BAT%d_T_BETA", index);
+	_analog_param_handles.t_beta = param_find(param_name);
 }
 
 void
-AnalogBattery::updateBatteryStatusADC(hrt_abstime timestamp, float voltage_raw, float current_raw)
+AnalogBattery::updateBatteryStatusADC(hrt_abstime timestamp, float voltage_raw, float current_raw,
+				      float temperature_raw)
 {
 	const float voltage_v = voltage_raw * _analog_params.v_div;
 	const float current_a = (current_raw - _analog_params.v_offs_cur) * _analog_params.a_per_v;
 
+	static constexpr float zero_c_in_k = 273.15f;
+	static constexpr float default_v_src = 5.f;
+
+	float v_src = default_v_src;
+	system_power_s system_power{};
+
+	if (_system_power_sub.copy(&system_power)) {
+		v_src = system_power.voltage5v_v;
+	}
+
+	float temperature_c = NAN;
+
+	if (std::fabsf(temperature_raw) > 0.001f && std::fabsf(v_src - temperature_raw) > 0.001f) {
+		const float r_thermistor = _analog_params.t_r_pu / ((v_src / temperature_raw) - 1.f);
+		const float r_inf = _analog_params.t_r_25c * expf(-_analog_params.t_beta / (zero_c_in_k + 25.f));
+		temperature_c = _analog_params.t_beta / logf(r_thermistor / r_inf) - zero_c_in_k;
```

Comment:
I think this can crash if `_analog_params.t_r_pu < 0` or `_analog_params.t_r_25c < 0`, then `logf` gets negative argument. IDK what that function does then

Reply:
Could be easier to verify if we split into smaller variables and validate them each before used in a dangerous function

Reply:
Then I would go with just one of two paradigms:
- Don't do input validation, let the value become nan (probably there is some "divide or nan" function if it's not already the result of division by zero)
- Only use validated input